azurerm/internal/services/containers/resource_arm_kubernetes_cluster.go

```diff
@@ -789,6 +783,29 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 	// since there's multiple reasons why we could be called into Update, we use this to only update if something's changed that's not SP/Version
 	updateCluster := false
 
+	// RBAC profile updates need to be handled atomically before any call to createUpdate as a diff there will create a PropertyChangeNotAllowed error
+	if d.HasChange("role_based_access_control") {
+		rbacRaw := d.Get("role_based_access_control").([]interface{})
+		rbacEnabled, azureADProfile := expandKubernetesClusterRoleBasedAccessControl(rbacRaw, tenantId)
+		// changing rbacEnabled must still force cluster recreation
+		if *existing.ManagedClusterProperties.EnableRBAC == rbacEnabled {
```

Comment:
I think it would be cleaner and more consistent with the rest of the provider to go:
```suggestion
		if props := existing.ManagedClusterProperties; if *props.EnableRBAC == rbacEnabled {
			props.AadProfile = azureADProfile
			...
```

also can we assume the deference will never be nil here?

Reply:
It defaults to 0 if not set, but that's not to say the API won't give us a nil on read somehow.  Good call.